Fix flow error in mapAsyncIterator

robzhu opened the pull request:
https://github.com/graphql/graphql-js/issues/1011

This fix attempts to address the flow errors found in libraries that depend on GraphQL-js but do not have `experimental.const_params=true` in their .flowconfig. 

With this explicit null check, I removed the `experimental.const_params=true` option from .flowconfig and saw all tests still pass.

wincent commented on src/subscription/mapAsyncIterator.js:
```diff
@@ -39,8 +39,9 @@ export default function mapAsyncIterator<T, U>(
   if (rejectCallback) {
-    mapReject = error =>
-      asyncMapValue(error, rejectCallback).then(iteratorResult, abruptClose);
+    mapReject = error => rejectCallback ?
```
A comment here would be helpful. Without `git blame`-ing this to see why the check is there, it's not obvious that it's for Flow.

robzhu commented:
Added a comment and replaced the redundant null check with a const variable to capture rejectCallback. 

> You mentioned changing the .flowconfig. Should that change also be included in this commit?

No, the change to .flowconfig was only to simulate running flow from dependent projects.

wincent commented:
> No, the change to .flowconfig was only to simulate running flow from dependent projects.

I meant though, not having that config means we're more likely to catch problems in dependent projects, at least for a while.

IvanGoncharov commented:
@wincent Can you please release `0.11.2` and update dependency in `express-graphql`?

robzhu commented:
> I meant though, not having that config means we're more likely to catch problems in dependent projects, at least for a while.

I'm fine with removing it to detect possible breakage in this repo early. @asiandrummer, you added that line in https://github.com/graphql/graphql-js/pull/885, any opposition/context?. In my branch, if I remove it, tests/flow still pass.

wincent commented:
@IvanGoncharov: v0.11.2 is out. express-graphql update should be out in a few minutes.

asiandrummer commented:
@robzhu @wincent I think I added the flag to fix the flow errors from Nuclide, for the exactly same reasons with why this PR exists (Nuclide had this flag, and graphql-js didn't). I think we should still have this flag, as it allows for a more strict flow check.
